src/useFetchye.js

```diff
@@ -14,48 +14,67 @@
  * permissions and limitations under the License.
  */
 
-import { useContext, useEffect, useRef } from 'react';
+import { useEffect, useRef } from 'react';
 import { runAsync } from './runAsync';
-import { defaultFetcher } from './defaultFetcher';
 import { computeKey } from './computeKey';
-import { isLoading } from './isLoading';
-import { FetchyeContext } from './FetchyeContext';
-import { getCacheByKey } from './getCacheByKey';
+import {
+  isLoading,
+} from './queryHelpers';
+import { useFetchyeContext } from './useFetchyeContext';
 import { defaultMapOptionsToKey } from './defaultMapOptionsToKey';
 
+const passInitialData = (value, initialValue, numOfRenders) => (numOfRenders === 1
+  ? value || initialValue
+  : value);
+
 export const useFetchye = (
   key,
   { mapOptionsToKey = (options) => options, ...options } = { },
-  fetcher = defaultFetcher) => {
+  fetcher) => {
   const {
-    useFetchyeSelector, dispatch, fetchClient,
-  } = useContext(FetchyeContext);
-  const cache = useFetchyeSelector();
+    defaultFetcher, useFetchyeSelector, dispatch, fetchClient,
+  } = useFetchyeContext();
+  const selectedFetcher = typeof fetcher === 'function' ? fetcher : defaultFetcher;
   const computedKey = computeKey(key, defaultMapOptionsToKey(mapOptionsToKey(options)));
```

Comment:
Would we need to compute this on every render?

Reply:
I just realized, it's not part of the changes...

Reply:
To answer your question with another question: If the key is an object, how do I know it changed internally but the parent reference remains equal?